public/js/ClusterPlot.js

```diff
@@ -0,0 +1,166 @@
+var d3 = require('d3');
+require('d3-tip')(d3);
+
+function ClusterPlot(sampleData) {
+    this.clusterMin = 2;
+    this.clusterMax = 20;
+    this.clusterMid = Math.round((this.clusterMax+this.clusterMin)/2);
+    this.sampleData = sampleData;
+
+    this.fullWidth = 650;
+    this.fullHeight = 450;
```

Comment:
I've been trained to despise magic numbers like this... Any way to define them elsewhere? e.g. Why is this not a particular class of plot that has certain stylings as defined here?